oauth2_http/java/com/google/auth/oauth2/UserAuthorizer.java

```diff
@@ -291,9 +320,15 @@ public UserCredentials getCredentialsFromCode(
     GenericData tokenData = new GenericData();
     tokenData.put("code", code);
     tokenData.put("client_id", clientId.getClientId());
-    tokenData.put("client_secret", clientId.getClientSecret());
     tokenData.put("redirect_uri", resolvedCallbackUri);
     tokenData.put("grant_type", "authorization_code");
+    // If the token request is sent to STS. STS will fail by design if the `client_secret` in post
+    // body.
+    // Sending client_secret when the client authentication type is post is to maintain existing
+    // behavior for existing code.
```

Comment:
Since this is a public method, can you add this behavior to the javadocs above? i.e. something along the lines of `No validation for uri with clientAuthorizationType and may result in failure` (but worded clearer)

Reply:
No problem. Moved to javadoc